src/components/sbb-expansion-panel/sbb-expansion-panel.tsx

```diff
@@ -0,0 +1,280 @@
+    const elements = (event.target as HTMLSlotElement).assignedElements();
+
+    // Changing titleLevel sometimes triggers a slot change with no assigned elements.
+    if (!elements.length) {
+      return;
+    }
+
+    this._headerRef = elements.find(
+      (e): e is HTMLSbbExpansionPanelHeaderElement => e.tagName === 'SBB-EXPANSION-PANEL-HEADER',
+    );
+
+    if (!this._headerRef) {
+      return;
+    }
+
+    this._headerRef.setAttribute('aria-expanded', String(this.expanded));
+    if (this.disabled) {
+      this._headerRef.setAttribute('disabled', String(this.disabled));
+    }
+    this._linkHeaderAndContent();
+  }
+
+  private _onContentSlotChange(event): void {
+    const elements = (event.target as HTMLSlotElement).assignedElements();
+
+    if (!elements.length) {
+      return;
+    }
+
+    if (this._contentRef) {
+      this._panelContentResizeObserver.unobserve(this._contentRef);
+    }
+    this._transitionEventController?.abort();
+
+    this._contentRef = (event.target as HTMLSlotElement)
+      .assignedElements()
+      .find(
+        (e): e is HTMLSbbExpansionPanelContentElement =>
+          e.tagName === 'SBB-EXPANSION-PANEL-CONTENT',
+      );
+
+    if (!this._contentRef) {
+      return;
+    }
+
+    this._panelContentResizeObserver.observe(this._contentRef);
+
+    this._transitionEventController = new AbortController();
+    this._contentRef.setAttribute('aria-hidden', String(!this.expanded));
+    this._contentRef.addEventListener('transitionend', (event) => this._onTransitionEnd(event), {
+      signal: this._transitionEventController.signal,
+    });
+    this._linkHeaderAndContent();
+  }
+
+  private _linkHeaderAndContent(): void {
+    if (!this._headerRef || !this._contentRef) {
+      return;
+    }
+
+    if (!this._headerRef.id) {
+      this._headerRef.setAttribute('id', `sbb-expansion-panel-header${this._progressiveId}`);
+    }
+    this._headerRef.setAttribute(
+      'aria-controls',
+      this._contentRef.id || `sbb-expansion-panel-content${this._progressiveId}`,
+    );
+
+    if (!this._contentRef.id) {
+      this._contentRef.setAttribute('id', `sbb-expansion-panel-content${this._progressiveId}`);
+    }
+    this._contentRef.setAttribute(
+      'aria-labelledby',
+      this._headerRef.id || `sbb-expansion-panel-header${this._progressiveId}`,
+    );
+    toggleDatasetEntry(this._contentRef, 'iconSpace', this._headerRef.hasAttribute('data-icon'));
+  }
+
+  private _onTransitionEnd(event): void {
+    // All transitions have the same timing and opacity is defined last, be sure that they have all been performed.
+    if (event.propertyName !== 'opacity') {
+      return;
+    }
+
+    if (this.expanded) {
+      this._onOpened();
+    } else {
+      this._onClosed();
+    }
+  }
+
+  private _onPanelContentResize(): void {
+    if (!this.expanded) {
+      return;
+    }
+
+    clearTimeout(this._resizeObserverTimeout);
+
+    this._resizeDisableAnimation = true;
```

Comment:
Is there a specific reason why this is done via state property and not directly via data attribute API?

Reply:
I assume to force a re-render? 
@federicoisepponfincons did Jeri discuss it with you?

Reply:
Hi guys! I agree with Lukas, it can probably be done via data attribute. I also quickly tested it and it should work. Thanks for the suggestion 😁

Reply:
Maybe it would be nice, if we adapt both implementations (notification and accordion) so they align to each other?

Reply:
Yep, I would do so. Shall I create a new PR for the notification or shall we change it in this PR as well?

Reply:
@DavideMininni-Fincons Marco has created the separate pr for notification, can you align the implementation in accordion?

Reply:
Yes, done in 8470a5bd07cc586e3e6c33917ef93e3ae51725d0